php/libraries/NDB_Page.class.inc

```diff
@@ -927,6 +927,110 @@ class NDB_Page implements RequestHandlerInterface
         );
     }
 
+    /**
+     * Get the list of projects for which the current user has permission
+     * To populate the module projects filter options
+     *
+     * @param bool $numeric_keys If the returned projects keys
+     *                           should be numeric (ProjectID)
+     *                           or strings (project name)
+     *
+     * @return array - an associative array of the form (ID => name)
+     */
+    public function getProjectOptions(
+        bool $numeric_keys = true
+    ): array {
+        $projectsOptions = [];
+        $userProjectsIDs = \NDB_Factory::singleton()->user()->getProjectIDs();
```

Comment:
This looks like it would make more sense as a parameter, but if it's a parameter I don't see why this function should be in the NDB_Page class at all. It would make more sense to have a `getProjects` method to the `User` class that returns an iterable of `\Project` instances. It could then be massaged into any format wanted by `array_map`/`array_values`/etc. A user's projects doesn't really have anything to do with the page.